Add netpol endport documentation (#26714)
kubernetes/website: content/en/docs/concepts/services-networking/network-policies.md

You must be using a {{< glossary_tooltip text="CNI" term_id="cni" >}} plugin that supports SCTP protocol NetworkPolicies.
{{< /note >}}

## Targeting a range of Ports

{{< feature-state for_k8s_version="v1.21" state="alpha" >}}

> When writing a Network Policy, you can target a range of Ports instead of a single port.

Reviewer: nit:
```suggestion
When writing a NetworkPolicy, you can target a range of ports instead of a single port.
```

This is achiveable with the usage of the `endPort` field, as the following example:

```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: multi-port-egress
  namespace: default
spec:
  podSelector:
    matchLabels:
      role: db
  policyTypes:
  - Egress
  egress:
  - to:
    - ipBlock:
        cidr: 10.0.0.0/24
    ports:
    - protocol: TCP
      port: 32000
      endPort: 32768
```

The above rule will allow a Pod with label `db` on the namespace `default` to communicate with any IP within the range `10.0.0.0/24` if the target port is between the range 32000 and 32768.

The following restrictions apply when using this field:
* As an alpha feature, this is disabled by default. To enable endPort field at a cluster level, you (or your cluster administrator) will need to enable the `NetworkPolicyEndPort` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) for the API server with `--feature-gates=NetworkPolicyEndPort=true,…`.
* The `endPort` field must be equal than or greater to `port` field.
* `endPort` can only be defined if `port` is also defined.

> * When using `endPort` field, the `port` field must be numeric.

Reviewer 2: maybe bringing a more global statement like `namedPorts are not supported` or both ports must be numeric.

Author: makes sense, will improve this statement during the weekend

{{< note >}}

> You must be using a {{< glossary_tooltip text="CNI" term_id="cni" >}} plugin that supports endPort field in NetworkPolicies ports specification.

Reviewer: ```suggestion
Your cluster must be using a {{< glossary_tooltip text="CNI" term_id="cni" >}} plugin that
supports the `endPort` field in NetworkPolicy specifications.
```
?
